src/lib/mailer/index.js

```diff
@@ -73,6 +73,21 @@ export const frequencyToInterval = (freq) => {
   return date;
 };
 
+/**
+ * Filters and deduplicates an array of email addresses, removing duplicates
+ * and excluding email addresses that start with 'no-send_'.
+ *
+ * @param {string[]} emails - An array of email addresses to filter and deduplicate.
+ * @returns {string[]} - A deduplicated and filtered array of email addresses.
+ */
+export const filterAndDeduplicateEmails = (emails) => {
+  const filteredEmails = emails.flat()
+    .filter((email) => typeof email === 'string' && !email.startsWith('no-send_'))
+    .filter((email, index, array) => array.indexOf(email) === index);
```

Comment:
Is this faster then building an serializing a set?

Reply:
I initially began with a set implementation but then switched to filtering. This method is one of the simplest ways to achieve both filtering and deduplication. While its performance implications are uncertain, considering our dataset consists of a very small number of email addresses, any performance difference should be negligible.